tests/integration/targets/ndo_tenant_mld_snooping_policy/tasks/main.yml

```diff
@@ -0,0 +1,401 @@
+          - nm_update_mld_snooping_policy_again.current.enableAdminState == "enabled"
+          - nm_update_mld_snooping_policy_again.current.enableFastLeaveControl == true
+          - nm_update_mld_snooping_policy_again.current.enableQuerierControl == true
+          - nm_update_mld_snooping_policy_again.current.lastMemberQueryInterval == 2
+          - nm_update_mld_snooping_policy_again.current.mldQuerierVersion == "v1"
+          - nm_update_mld_snooping_policy_again.current.queryInterval == 100
+          - nm_update_mld_snooping_policy_again.current.queryResponseInterval == 5
+          - nm_update_mld_snooping_policy_again.current.startQueryCount == 1
+          - nm_update_mld_snooping_policy_again.current.startQueryInterval == 25          
+          - nm_update_mld_snooping_policy_again.current.uuid is defined
+
+    - name: Update a MLD Snooping Policy name
+      cisco.mso.ndo_tenant_mld_snooping_policy: &update_mld_snooping_policy_name
+        <<: *update_mld_snooping_policy
+        uuid: '{{ nm_update_mld_snooping_policy.current.uuid }}'
+        name: ansible_test_mld_snooping_policy_changed
+      register: nm_update_mld_snooping_policy_name
+
+    - name: Assert MLD Snooping Policy was updated
+      assert:
+        that:
+          - nm_update_mld_snooping_policy_name is changed
+          - nm_update_mld_snooping_policy_name.previous.name == "ansible_test_mld_snooping_policy"
+          - nm_update_mld_snooping_policy_name.current.name == "ansible_test_mld_snooping_policy_changed"
+
+    # QUERY
+
+    - name: Create another MLD Snooping Policy
+      cisco.mso.ndo_tenant_mld_snooping_policy: &create_mld_snooping_policy_2
+        <<: *create_mld_snooping_policy
+        name: ansible_test_mld_snooping_policy_2
+        description: "This is a test MLD Snooping Policy 2"
+
+    - name: Query a MLD Snooping Policy with template_name
+      cisco.mso.ndo_tenant_mld_snooping_policy:
+        <<: *create_mld_snooping_policy_2
+        state: query
+      register: query_one
+
+    - name: Query a MLD Snooping Policy with template_name and UUID
+      cisco.mso.ndo_tenant_mld_snooping_policy: &query_mld_snooping_policy_uuid
+        <<: *create_mld_snooping_policy_2
+        uuid: '{{ query_one.current.uuid }}'
+        name: '{{ fakevar | default(omit)}}'
+        state: query
+      register: query_one_uuid
+
+    - name: Query all MLD Snooping Policy in the template
+      cisco.mso.ndo_tenant_mld_snooping_policy:
+        <<: *mso_info
+        template: ansible_tenant_template
+        state: query
+      register: query_all
+
+    - name: Assert MLD Snooping Policy was queried
+      assert:
+        that:
+          - query_one is not changed
+          - query_one.current.name == "ansible_test_mld_snooping_policy_2"
+          - query_one_uuid is not changed
+          - query_one_uuid.current.name == "ansible_test_mld_snooping_policy_2"
+          - query_all is not changed
+          - query_all.current | length == 2
```

Comment:
Should the test also check that the correct 2 policies are being returned?

Reply:
we do not do in other will add here, also here do you want to create an issue to do this for all other modules?

Reply:
Not sure if we need to retrospectively change the tests now. I think its okay to leave them but I have been asking everyone to do this in new tests.